src/Bicep.Core/Semantics/SymbolExtensions.cs

```diff
@@ -81,5 +81,11 @@ public static TypeSymbol GetDeclaredArgumentType(this IFunctionSymbol functionSy
 
             return TypeHelper.CreateTypeUnion(argTypes);
         }
+
+        /// <summary>
+        ///   Certain declarations (outputs and metadata) define symbols which can't be referenced by name. This method allows you to filter out non-referencable symbols.
+        /// </summary>
+        public static bool CanBeReferenced(this DeclaredSymbol declaredSymbol)
+            => declaredSymbol is not OutputSymbol and not MetadataSymbol;
```

Comment:
>MetadataSymbol [](http://example.com/codeflow?start=58&length=14)

Should we make this an interface or a property on some base type instead of hard-coding? (The list of these keeps growing.)

Reply:
I'm going to follow the [rule of 3](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) here and leave this problem up to the next person 😛. Definitely has nothing to do with me being too lazy to start a new iteration...

Reply:
Haha fair enough.